browser/ui/webui/instant_service_message_handler.cc

```diff
@@ -93,24 +99,27 @@ void InstantServiceMessageHandler::MostVisitedInfoChanged(
   base::Value tiles(base::Value::Type::LIST);
   int tile_id = 1;
 
-  auto* service = ViewCounterServiceFactory::GetForProfile(profile_);
-  for (auto& top_site : service->GetTopSitesVectorForWebUI()) {
-    base::Value tile_value(base::Value::Type::DICTIONARY);
-    if (top_site.name.empty()) {
-      tile_value.SetStringKey("title", top_site.destination_url);
-      tile_value.SetIntKey("title_direction", base::i18n::LEFT_TO_RIGHT);
-    } else {
-      tile_value.SetStringKey("title", top_site.name);
-      tile_value.SetIntKey("title_direction",
-          base::i18n::GetFirstStrongCharacterDirection(
-              base::UTF8ToUTF16(top_site.name)));
+  // Super Referral feature only present in regular tabs (not private tabs)
+  if (!IsPrivateNewTab(profile_)) {
```

Comment:
There is already a shared util function in profile_util.h, `!brave::IsRegularProfile(profile)` is what you want here I believe.
Though in this particular case, I think it's better to check if the returned service is null and early return if so, we already have logic in ViewCounterServiceFactory to decide whether it is a supported profile, we shouldn't need to duplicate the logic here again.

Reply:
Fixed!